From fcb4e8d261573ee49ee20f0e418a07f7e16403c8 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Wed, 16 Sep 2026 08:22:35 +0800
Subject: [PATCH 44/63] UPSTREAM: x86-64: Link tst-shstk-legacy-1{f,g} with
 -Wl,--no-as-needed

Link tst-shstk-legacy-1f and tst-shstk-legacy-1g with -Wl,--no-as-needed
so that tst-shstk-legacy-mod-1.so is added to DT_NEEDED.  This fixes
BZ #31877.

Signed-off-by: H.J. Lu <hjl.tools@gmail.com>
Reviewed-by: Aurelien Jarno <aurelien@aurel32.net>
(cherry picked from commit 1db83d1652d35115525960021e4fb5b76dd1c633)
---
 sysdeps/x86_64/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sysdeps/x86_64/Makefile b/sysdeps/x86_64/Makefile
index ff0463d7ca..dfb344d74a 100644
--- a/sysdeps/x86_64/Makefile
+++ b/sysdeps/x86_64/Makefile
@@ -467,10 +467,12 @@ $(objpfx)tst-shstk-legacy-1e-static.out: \
 	$(SHELL) $< $(common-objpfx) 2> $@; \
 	$(evaluate-test)
 tst-shstk-legacy-1f-ENV = GLIBC_TUNABLES=glibc.cpu.hwcaps=SHSTK
+LDFLAGS-tst-shstk-legacy-1f = -Wl,--no-as-needed
 $(objpfx)tst-shstk-legacy-1f: $(objpfx)tst-shstk-legacy-mod-1.so
 $(objpfx)tst-shstk-legacy-mod-1.so: \
   $(objpfx)tst-shstk-legacy-mod-1.os \
   $(objpfx)tst-shstk-legacy-1-extra.os
+LDFLAGS-tst-shstk-legacy-1g = -Wl,--no-as-needed
 $(objpfx)tst-shstk-legacy-1g: $(objpfx)tst-shstk-legacy-mod-1.so
 $(objpfx)tst-shstk-legacy-1g.out: \
   $(..)/sysdeps/x86_64/tst-shstk-legacy-1g.sh $(objpfx)tst-shstk-legacy-1g
-- 
2.55.0

